### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mkucherenko**
Fixes for tests and generating url

### CODE REVIEW

File: tests/Tokenio/TokenBaseTest.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use Io\Token\Proto\Common\Address\Address;
+use PHPUnit\Framework\TestCase;
+use Tokenio\Config\TokenCluster;
+use Tokenio\Config\TokenEnvironment;
+use Tokenio\Config\TokenIoBuilder;
+use Tokenio\Util\Strings;
+use Tokenio\Security\UnsecuredFileSystemKeyStore;
+use Io\Token\Proto\Common\Alias\Alias;
+
+abstract class TokenBaseTest extends TestCase
```
Comment (yuhanniu) [inline tests/Tokenio/TokenBaseTest.php]: TestUtil

File: tests/Tokenio/MemberRegistrationTest.php [hunk 2]
```diff
@@ -16,9 +16,13 @@ class MemberRegistrationTest extends TestCase
     /** @var \Tokenio\TokenIO */
     protected $tokenIO;
 
+    /** @var UnsecuredFileSystemKeyStore */
+    protected $keyStore;
+
     protected function setUp()
     {
         $this->tokenIO = TestUtil::initializeSDK();
+        $this->keyStore = new UnsecuredFileSystemKeyStore(__DIR__ . '/test-keys/');
```
Comment (yuhanniu) [inline tests/Tokenio/MemberRegistrationTest.php]: the key store is only used in one test, shouldn't be in the `setUp()`

File: tests/Tokenio/MemberRegistrationTest.php [hunk 3]
```diff
@@ -137,4 +141,44 @@ public function testRecovery()
         $this->assertEquals([$alias], $actualAliases);
     }
 
+    public function testRecoveryWithSecondaryAgent()
+    {
+        $this->setUp();
```
Comment (yuhanniu) [inline tests/Tokenio/MemberRegistrationTest.php]: setup is automatically called right?

File: tests/Tokenio/ProfileTest.php [hunk 4]
```diff
@@ -63,9 +63,50 @@ public function testReadProfile_notYours()
         $this->assertEquals($inProfile, $outProfile);
     }
 
+    public function testGetProfilePicture()
+    {
+        $this->setUp();
```
Comment (yuhanniu) [inline tests/Tokenio/ProfileTest.php]: same

File: tests/Tokenio/MemberRegistrationTest.php (hunk 3 above)
Comment (mkucherenko) [reply tests/Tokenio/MemberRegistrationTest.php]: That was added for testing purpose, I'll remove that

File: tests/Tokenio/TokenBaseTest.php (hunk 1 above)
Comment (mkucherenko) [reply tests/Tokenio/TokenBaseTest.php]: BaseClass removed

File: lib/Http/UnauthenticatedClient.php [hunk 5]
```diff
@@ -5,6 +5,7 @@
 use Io\Token\Proto\Common\Alias\Alias;
 use Io\Token\Proto\Common\Alias\VerificationStatus;
 use Io\Token\Proto\Common\Member\Member;
+use Io\Token\Proto\Common\Member\MemberAddKeyOperation;
```
Comment (yuhanniu) [inline lib/Http/UnauthenticatedClient.php]: rm

File: lib/Http/UnauthenticatedClient.php [hunk 6]
```diff
@@ -16,6 +17,7 @@
 use Io\Token\Proto\Gateway\BeginRecoveryResponse;
 use Io\Token\Proto\Gateway\CompleteRecoveryRequest;
 use Io\Token\Proto\Gateway\CompleteRecoveryResponse;
+use Io\Token\Proto\Common\Transaction\RequestStatus;
```
Comment (yuhanniu) [inline lib/Http/UnauthenticatedClient.php]: rm

File: tests/Tokenio/TokenRequestTest.php [hunk 7]
```diff
@@ -27,20 +27,22 @@ protected function setUp()
 
     public function testAddAndGetTransferTokenRequest()
     {
-        $storedRequest = new TokenRequest();
+
```
Comment (yuhanniu) [inline tests/Tokenio/TokenRequestTest.php]: rm

**Pull request merged**